src/libsystemd-network/test-ipv4ll.c

```diff
@@ -168,6 +170,13 @@ static void test_basic_request(sd_event *e) {
 
                 sd_event_run(e, UINT64_MAX);
                 assert_se(basic_request_handler_bind == 1);
+
+                if (start_address->s_addr != 0) {
```

Comment:
Maybe, `in4_addr_is_link_local()` is better?

Reply:
That's sounds good